File: haskell/anagram/Anagram.hs
module Anagram (anagramsFor) where

import Data.List
import Data.Char

isAnagramFor :: String -> String -> Bool
isAnagramFor x y = lx /= ly && sort lx == sort ly
    where lx = map toLower x
          ly = map toLower y

anagramsFor :: String -> [String] -> [String]
anagramsFor s = filter (isAnagramFor s)

File: clojure/anagram/anagram.clj
(ns anagram)

(require '[clojure.string :as s])

(defn anagram? [str1 str2]
  (let [s1 (s/lower-case str1)
        s2 (s/lower-case str2)]
    (and (not= s1 s2)
         (= (sort s1) (sort s2)))))

(defn anagrams-for [str strs]
  (filter (fn [x] (anagram? str x)) strs))
